client/devdocs/design/search-collection.jsx

```diff
@@ -53,12 +53,19 @@ const Collection = ( {
 
 		const exampleName = getComponentName( example );
 		const exampleLink = `/devdocs/${ section }/${ camelCaseToSlug( exampleName ) }`;
-		const readmeFilePath =
-			'/client/' +
-			( section === 'blocks' ? 'blocks' : 'components' ) +
-			'/' +
-			example.props.readmeFilePath +
-			'/README.md';
+		let readmeFilePath = null;
+		let showEditLink = true;
+
+		if ( 'design' === section ) {
+			readmeFilePath = `/client/components/${ example.props.readmeFilePath }/README.md`;
+		} else if ( 'gutenberg-components' === section ) {
+			readmeFilePath = `/node_modules/@wordpress/components/src/${
+				example.props.readmeFilePath
+			}/README.md`;
+			showEditLink = false;
+		} else {
+			readmeFilePath = `/client/${ section }/${ example.props.readmeFilePath }/README.md`;
+		}
```

Comment:
Maybe pull this out into a helper functions?
```jsx
const shouldShowEditLink = ( section ) => ( section !== 'gutenberg-components' );

const getReadmeFilePath( section, example ) {
        switch( section ) {
             case 'design':
                 return `/client/components/${ example.props.readmeFilePath }/README.md`;
             case 'gutenberg-components';
                 return `/node_modules/@wordpress/components/src/${ example.props.readmeFilePath }/README.md`;
             default:
                  return `/client/${ section }/${ example.props.readmeFilePath }/README.md`;
        }
}
```

Reply:
Agreed!